identitatem/dex-operator
# Handle updates to the IDP client secret

If the IDP client secret (Github client secret, LDAP bind password) is changed, the DexServer should pick up the updated secrets to connect to the respective IDP.

To narrow the scope of the secrets being watched, the label `auth.identitatem.io/idp-credential=<any value>` needs to be present on the secret containing the Github Client secret.

To test:
1. Set up the DexServer, client and Oauth manifests as provided in the `hack/generate_cr.sh` script. (The Github client secret has the label `auth.identitatem.io/idp-credential` on it.
2. Install the operator and apply the manifests as described in the Readme 
3. Check that the Dex Server pod is created
4. Now edit the Github client secret - using `oc edit secret <GH client secret name>`. Change the "clientSecret" value within the data field.
5. When the edit is complete, a reconcile loop should be triggered. A new dex server deployment is rolled out with the updated secret.
6. You can check this with `oc get cm dex2`. The config map should have the updated client secret.
7. Try changing a different secret - such as the LDAP bind password/ Microsoft client secret. These secrets do not have the label `auth.identitatem.io/idp-credential` applied to them in the `generate_cr.sh` script, so edits on these shoould not trigger a reconcile loop.
